### PR TITLE
Debian11 with gcc10

### PR DESCRIPTION
**Pull request by shuston**
Work in progress.

Bringing a customer's changes in to current dev to support Debian 11 w/ gcc10.

### CODE REVIEW

**Comment by jwillemsen** [comment]
Steve, can you do a global search in ACE/TAO for any fallthrough comments and replace that with `ACE_FALLTHROUGH` as part of this PR, I think there are a few more places where we have used that comment to silence a warning.

**Comment by shuston** [comment]
Hi Johnny,

I will check through ACE, but I am not familiar enough with TAO to get into that.

-Steve

**Comment by jwillemsen** [comment]
Hi,

> Hi Johnny, I will check through ACE, but I am not familiar enough with TAO to get into that.

Looks a global find/replace will probably work, I will do that on TAO after you have finished and merged this PR

File: ACE/ace/Name_Proxy.cpp [hunk 1]
```diff
@@ -143,55 +143,55 @@ ACE_Name_Proxy::recv_reply (ACE_Name_Request &reply)
...
+			 n,
+			 sizeof (ACE_UINT32)));
+	}
+      // If 0, no diagnostic - the peer shut it down.
+      // We've shutdown unexpectedly
+      return -1;
+    }
+
+  // Transform the length into host byte order.
+  ssize_t length = ACE_NTOHL (reply.length ());
```
Comment (jwillemsen) [inline ACE/ace/Name_Proxy.cpp:169]: length as const?

File: ACE/ace/Time_Value.h [hunk 2]
```diff
@@ -67,6 +67,9 @@ class ACE_Export ACE_Time_Value
   /// Constructor.
   explicit ACE_Time_Value (time_t sec, suseconds_t usec = 0);
 
+  /// Assign @ tv to this
+  ACE_Time_Value (const ACE_Time_Value &tv);
```
Comment (jwillemsen) [inline ACE/ace/Time_Value.h:71]: Why does the default version not work? When there is a use-case where the default doesn't work we should add a test case for that

File: ACE/ace/Handle_Set.h [hunk 3]
```diff
@@ -71,6 +71,20 @@ class ACE_Export ACE_Handle_Set
    */
   ACE_Handle_Set (const fd_set &mask);
 
+
```
Comment (jwillemsen) [inline ACE/ace/Handle_Set.h:74]: Did something change in Handle_Set.h/inl, looking at the diffs it only looks some operations got moved in the file itself?

File: ACE/ace/Atomic_Op.h [hunk 4]
```diff
@@ -304,6 +304,7 @@ class ACE_Export ACE_Atomic_Op<ACE_Thread_Mutex, unsigned long>
   ACE_Atomic_Op (unsigned long c);
   ACE_Atomic_Op (const ACE_Atomic_Op<ACE_Thread_Mutex, unsigned long> &c);
   ACE_Atomic_Op<ACE_Thread_Mutex, unsigned long> &operator= (unsigned long rhs);
+  ACE_Atomic_Op<ACE_Thread_Mutex, unsigned long> &operator= (const ACE_Atomic_Op<ACE_Thread_Mutex, unsigned long> &c);
```
Comment (jwillemsen) [inline ACE/ace/Atomic_Op.h:307]: Some test extension for this operation? With C++11 we do prefer std::atomic, we should be careful extending ACE_Atomic_Op, not that we add all kinds of ACE extensions which make it harder later for users to migrate to std::atomic

**Review by mitza-oci — CHANGES_REQUESTED** [review]

File: ACE/ace/Handle_Set.inl [hunk 5]
```diff
@@ -24,10 +24,10 @@ ACE_Handle_Set::reset ()
 }
 
 #if defined (ACE_HAS_BIG_FD_SET)
-ACE_INLINE ACE_Handle_Set &
-ACE_Handle_Set::operator = (const ACE_Handle_Set &rhs)
+ACE_INLINE
+ACE_Handle_Set::ACE_Handle_Set (const ACE_Handle_Set &rhs)
```
Comment (mitza-oci) [inline ACE/ace/Handle_Set.inl:28]: This is changing the order of the copy constructor and copy assignment operator (see line 43) which makes the diff harder to read.  Is the order significant?

File: ACE/tests/Naming_Test.cpp [hunk 6]
```diff
@@ -263,7 +263,7 @@ run_main (int argc, ACE_TCHAR *argv[])
                        ACE_TEXT ("%s%d"),
 #endif
                        pname,
-                       (int)(ACE_OS::getpid ()));
+                       ACE_OS::getpid ());
```
Comment (mitza-oci) [inline ACE/tests/Naming_Test.cpp:266]: Is `pid_t` larger than int on any current (or future?) platform?

**Comment by jwillemsen** [comment]
@shuston Steve, the `pid_t` related change results in a new warning with mingw64, see https://buildlogs.remedy.nl/win_mingw64_64_tao/2022_04_24_07_04_Full.html#warning_137, can you have a look?

```
[[Details](https://buildlogs.remedy.nl/win_mingw64_64_tao/2022_04_24_07_04_Full.html#warning_137)] x:/BUILD/DOC_ROOT/ACE/tests/Naming_Test.cpp: In function 'int run_main(int, ACE_TCHAR**)':
[[Details](https://buildlogs.remedy.nl/win_mingw64_64_tao/2022_04_24_07_04_Full.html#warning_138)] x:/BUILD/DOC_ROOT/ACE/tests/Naming_Test.cpp:263:34: warning: format '%d' expects argument of type 'int', but argument 5 has type 'pid_t' {aka 'long long int'} [-Wformat=]
[[Details](https://buildlogs.remedy.nl/win_mingw64_64_tao/2022_04_24_07_04_Full.html#warning_139)] X:/BUILD/DOC_ROOT/ACE/ace/ace_wchar.h:97:27: note: in definition of macro 'ACE_TEXT'
```

**Comment by shuston** [comment]
> @shuston Steve, the `pid_t` related change results in a new warning with mingw64, see https://buildlogs.remedy.nl/win_mingw64_64_tao/2022_04_24_07_04_Full.html#warning_137, can you have a look?
> 
> ```
> [[Details](https://buildlogs.remedy.nl/win_mingw64_64_tao/2022_04_24_07_04_Full.html#warning_137)] x:/BUILD/DOC_ROOT/ACE/tests/Naming_Test.cpp: In function 'int run_main(int, ACE_TCHAR**)':
> [[Details](https://buildlogs.remedy.nl/win_mingw64_64_tao/2022_04_24_07_04_Full.html#warning_138)] x:/BUILD/DOC_ROOT/ACE/tests/Naming_Test.cpp:263:34: warning: format '%d' expects argument of type 'int', but argument 5 has type 'pid_t' {aka 'long long int'} [-Wformat=]
> [[Details](https://buildlogs.remedy.nl/win_mingw64_64_tao/2022_04_24_07_04_Full.html#warning_139)] X:/BUILD/DOC_ROOT/ACE/ace/ace_wchar.h:97:27: note: in definition of macro 'ACE_TEXT'
> ```

Absolutely... apologies for that error, Johnny. @mitza-oci - you tried to warn me - sorry, I misunderstood what you were saying.

**Pull request merged**